### PR TITLE
Add action to create a module release PR

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: .github/workflows/prepare_release.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: true
+        type: string
+      allowed_owner:
+        description: 'Only allow this owner'
+        required: true
+        type: string
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: Since it's triggered explicitly, is this really needed? The reason we have it in the release is that you can push tags to your own fork and you don't want it to start releasing from there.

File: .github/workflows/prepare_release.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+    secrets:
+      username:
+        description: 'The forge username to use'
+        required: true
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: I don't see this secret used in the flow itself.

File: .github/workflows/prepare_release.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --local user.email "${{ github.repository_owner }}@users.noreply.github.com"
+          git config --local user.name "GitHub Actions"
+          git add .
+          git commit -m "Release v${{ steps.get_version.outputs.version }}"
+          gh pr create --title "Release v${{ steps.get_version.outputs.version }}" --label skip-changelog
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: In Foreman we use [peter-evans/create-pull-request](https://github.com/peter-evans/create-pull-request) (see https://github.com/theforeman/foreman-packaging/blob/605edf36c9cad3e062ce855601e52b93442c8fa8/.github/workflows/bump_packages.yml#L105-L112) and that has the nice benefit that it will update an existing PR by reusing a branch.

File: .github/workflows/prepare_release.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --local user.email "${{ github.repository_owner }}@users.noreply.github.com"
+          git config --local user.name "GitHub Actions"
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: In Foreman we set these via env vars: https://github.com/theforeman/foreman-packaging/blob/605edf36c9cad3e062ce855601e52b93442c8fa8/.github/workflows/bump_packages.yml#L8-L10

File: .github/workflows/prepare_release.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        run: bundle exec rake module:bump_to_version[${{ inputs.version }}]
+      - name: Update CHANGELOG.md and update REFERENCE.md if required
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: bundle exec rake release:prepare
+      - name: 'Check if a release is necessary'
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: Does this work? I think if you bump to a new version even if there aren't changes then it still creates an empty header for the new version.

File: .github/workflows/prepare_release.yml [hunk 6]
```diff
@@ -0,0 +1,67 @@
+---
+
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: true
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: I would love it if we had some logic to automatically detect this based on labels, making it optional but when I looked at it I had trouble coming up with a good solution so this is a good first step.

File: .github/workflows/prepare_release.yml (hunk 1 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: dropped it

File: .github/workflows/prepare_release.yml (hunk 2 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: dropped it

File: .github/workflows/prepare_release.yml (hunk 6 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: dito. I would also love to do this automatically. but one step at a time.

File: .github/workflows/prepare_release.yml (hunk 5 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: I don't know yet, I want to test it.

File: .github/workflows/prepare_release.yml (hunk 4 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: updated it to use env vars

File: .github/workflows/prepare_release.yml (hunk 5 above)
Comment (ekohl) [reply .github/workflows/prepare_release.yml]: I tested it manually in [puppet-dns](https://github.com/theforeman/puppet-dns) and it appears to work:
```console
$ git log 11.0.1..HEAD
$ be rake module:bump_to_version[11.0.2]
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
Bumping version to 11.0.2
Bumping version to 11.0.2
$ changelog
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
Found 41 tags                                                          
Fetching tags dates: 41/41
Sorting tags...
Received issues: 263                                                   
Pull Request count: 239                                                
Filtered pull requests: 124
Filtered issues: 12
Fetching events for issues and PR: 136                                 
Fetching closed dates for issues: 136/136
Associating PRs with tags: 124/124
Generating entry...
Done!
Generated log placed in /home/ekohl/dev/puppet-dns/CHANGELOG.md
$ git diff CHANGELOG.md
$ echo $?
0
```

File: .github/workflows/prepare_release.yml (hunk 3 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: I wanted to give `gh` a try. we already use it to create the actual github release. And I was hoping that we can avoid a 3rd party workflow.

File: .github/workflows/prepare_release.yml (hunk 5 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: ah great!

File: .github/workflows/prepare_release.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GIT_AUTHOR_NAME: Vox Pupuli Release Automation
+          GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: Can we set this at the top level?

File: .github/workflows/prepare_release.yml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      - name: 'Check if a release is necessary'
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GIT_AUTHOR_NAME: Vox Pupuli Release Automation
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: I wouldn't mind using this same action for Foreman and wonder if would want something else. Any thoughts on generalizing it?
```suggestion
          GIT_AUTHOR_NAME: Release Automation
```

Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: sure

File: .github/workflows/prepare_release.yml (hunk 7 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: moved GIT_AUTHOR_NAME / GIT_AUTHOR_EMAIL. I kept GH_TOKEN, to not expose the secret too much.

File: .github/workflows/prepare_release.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git commit --all --message="Release v${{ steps.get_version.outputs.version }}"
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: We never use the `v` prefix in the commit message
```suggestion
          git commit --all --message="Release ${{ steps.get_version.outputs.version }}"
```

File: .github/workflows/prepare_release.yml (hunk 3 above)
Comment (ekohl) [reply .github/workflows/prepare_release.yml]: The current workflow doesn't switch to another branch and probably commits straight to master.

Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: It creates a `release_prep` branch now. Perforce uses the same branch name.

File: .github/workflows/prepare_release.yml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Puppet Forge'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml:48]: With my yellow Foreman hat on: we maintain stable branches as well and if we need to create multiple release then it would be nice if there was a parameter for the base branch. Then you should also take that into account for the target branch name

File: .github/workflows/prepare_release.yml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git switch --create release_prep
+          git commit --all --message="Release ${{ steps.get_version.outputs.version }}"
+          gh pr create --title "Release ${{ steps.get_version.outputs.version }}" --label skip-changelog
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: How does `gh pr create` work if a PR already exists? Does it at least push to the existing branch or just error out? You know how common it's to create a prep, fix labels and then regenerate it.

File: .github/workflows/prepare_release.yml (hunk 10 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml:48]: updated it. it defaults to `master` now, but you can provide a different branch name as base  for the `release_prep` branch / for the PR.

File: .github/workflows/prepare_release.yml (hunk 11 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: I don't know yet. That's why I raised https://github.com/voxpupuli/puppet-example/pull/81 , to test it.

Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: when running it on code where a PR already exists:

```
$ gh pr create
a pull request for branch "prep" into branch "v3" already exists:
https://github.com/voxpupuli/gha-puppet/pull/66
```

File: .github/workflows/prepare_release.yml [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+        with:
+          ref: ${{ inputs.base-branch }}
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+          bundler-cache: true
+          working-directory: ${{ inputs.working-directory }}
+      - name: Update metadata.json to new version
+        run: bundle exec rake module:bump_to_version[${{ inputs.version }}]
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections recommends setting it to an environment variable instead of using shell interpolation. While the risk may be small here, I think it's still a good idea.

Perhaps we can even modify puppet-blacksmith to understand an environment variable. I'd be tempted to modify `module:bump` to support `BLACKSMITH_NEXT_VERSION` and otherwise simply bump a patch level. That would be around here: https://github.com/voxpupuli/puppet-blacksmith/blob/760c035c6b87861ca767358d96178f8ebdd6ec13/lib/puppet_blacksmith/rake_tasks.rb#L76-L88

And if you take that logic a step further, you can make the input version optional then. The workflow becomes:

* Call release prep without arguments
* Inspect the generated changelog
* Fix any labels
* Call the release workflow with the correct version number
* Merge PR
* Create tag

As a follow up: you could create a workflow that if the `release_prep` branch is merged that you create a tag automatically.

File: .github/workflows/prepare_release.yml (hunk 10 above)
Comment (ekohl) [reply .github/workflows/prepare_release.yml:48]: If you don't specify any default, it should just work regardless of the default branch.

File: .github/workflows/prepare_release.yml (hunk 3 above)
Comment (ekohl) [reply .github/workflows/prepare_release.yml]: Perforce doesn't maintain stable branches so they don't need it. If they did, we wouldn't have the whole stdlib 9.x debacle and could have easily created an 8.x release with the non-deprecated functions.

File: .github/workflows/prepare_release.yml (hunk 12 above)
Comment (ekohl) [reply .github/workflows/prepare_release.yml]: Turns out, there's already `module:bump:full` which uses `BLACKSMITH_FULL_VERSION`
```suggestion
        run: bundle exec rake module:bump:full
```

File: .github/workflows/prepare_release.yml [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+      base-branch:
+        description: 'The branch that will be used as the origin for the release branch.'
+        required: false
+        default: 'master'
+        type: string
+
+env:
+  BUNDLE_WITHOUT: development:test:system_tests
+  GIT_AUTHOR_NAME: Release Automation
+  GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml:35]: Tying into https://github.com/voxpupuli/gha-puppet/pull/66/files#r1892949324 I think this is needed
```suggestion
  GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
  BLACKSMITH_FULL_VERSION: "${{ inputs.version }}"
```

File: .github/workflows/prepare_release.yml (hunk 12 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: updated it

File: .github/workflows/prepare_release.yml (hunk 3 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: I know. But their release branch is always named `release_prep`, I simply wanted to copy that.

File: .github/workflows/prepare_release.yml [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git switch --create release_prep
+          git commit --all --message="Release ${{ steps.get_version.outputs.version }}"
+          gh pr create --title "Release ${{ steps.get_version.outputs.version }}" --label skip-changelog --base ${{ inputs.base-branch }} --fill
```
Comment (evgeni) [inline .github/workflows/prepare_release.yml]: I can only recommend using `peter-evans/create-pull-request` instead of wrangling all those steps manually.

Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: ok ok!

File: .github/workflows/prepare_release.yml [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+        run: bundle exec rake release:prepare
+      - name: 'Check if a release is necessary'
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (evgeni) [inline .github/workflows/prepare_release.yml]: One interesting side note: if that's the normal token GH generates on the fly when the action runs, the created PR will not trigger any further GH actions (like CI).

*Reads docs* https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow

Oh, this is not the case for workflow dispatch, TIL

File: .github/workflows/prepare_release.yml (hunk 14 above)
Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: updated it

File: .github/workflows/prepare_release.yml [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+        with:
+          ref: ${{ inputs.base-branch }}
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+          bundler-cache: true
+          working-directory: ${{ inputs.working-directory }}
+      - name: Update metadata.json to new version
+        run: bundle exec rake module:bump:full
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: One way to implement https://github.com/voxpupuli/gha-puppet/pull/66/files#r1892651518 is to bump the patch level if it isn't given. While it may not be perfect, it means you don't need to specify it which is commonly what you do on the first run.
```suggestion
        run: |
          if [[ -n BLACKSMITH_FULL_VERSION ]] ; then
            bundle exec rake module:bump:full
          else
            bundle exec rake module:bump
          fi
```

File: .github/workflows/prepare_release.yml [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+      - name: Update CHANGELOG.md and update REFERENCE.md if required
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]: Perhaps keep it generic and keep the implementation to voxpupuli-release?
```suggestion
      - name: Prepare the release
```

File: .github/workflows/prepare_release.yml [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+  GIT_AUTHOR_NAME: Release Automation
+  GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
+  BLACKSMITH_FULL_VERSION: "${{ inputs.version }}"
+
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Puppet Forge'
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml]:
```suggestion
    name: 'Prepare Release'
```

File: .github/workflows/prepare_release.yml (hunk 6 above)
Comment (ekohl) [reply .github/workflows/prepare_release.yml]: After https://github.com/voxpupuli/gha-puppet/pull/66#discussion_r1894705081 this can be
```suggestion
        required: false
```

File: .github/workflows/prepare_release.yml [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+          fi
+      - name: Prepare the release
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: bundle exec rake release:prepare
+      - name: 'Check if a release is necessary'
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
```
Comment (evgeni) [inline .github/workflows/prepare_release.yml]: Will there ever be the case where the changelog is unchanged, but some other file is changed?

If not, you can skip this check as the action will handle the "no changes" case fine.

https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#action-behaviour

File: .github/workflows/prepare_release.yml [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+      - name: Create pull Request
+        uses: peter-evans/create-pull-request@v7
+        if: ${{ steps.check.outputs.release == 'true' }}
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (evgeni) [inline .github/workflows/prepare_release.yml]: This is the default and can be omitted

File: .github/workflows/prepare_release.yml [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+        type: string
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+      base-branch:
+        description: 'The branch that will be used as the origin for the release branch.'
+        required: false
+        default: 'master'
```
Comment (evgeni) [inline .github/workflows/prepare_release.yml]: If you don't set a default, it will work with any default branch ;)

File: .github/workflows/prepare_release.yml [hunk 22]
```diff
@@ -0,0 +1,72 @@
...
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Prepare Release'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
```
Comment (bastelfreak) [inline .github/workflows/prepare_release.yml:49]: Perforce makes a full clone with:

```yaml
      - name: "Checkout"
        uses: "actions/checkout@v4"
        with:
          fetch-depth: 0
```

In my tests that wasn't required. Maybe it's for some edge cases. Based on their git history I didn't find any explanation.

**Comment by bastelfreak** [comment]
Tested successfully at https://github.com/voxpupuli/puppet-example/pull/82 . Followup PR for the optional input: https://github.com/voxpupuli/puppet-example/pull/83

**Review by jay7x — APPROVED** [review]

File: .github/workflows/prepare_release.yml (hunk 19 above)
Comment (ekohl) [reply .github/workflows/prepare_release.yml]: Metadata is always bumped, right?

Comment (bastelfreak) [reply .github/workflows/prepare_release.yml]: I think the whole check isn't required anymore when peter-evans/create-pull-request@v7 is used.

File: .github/workflows/prepare_release.yml [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+            bundle exec rake module:bump:full
+          else
+            bundle exec rake module:bump
+          fi
+      - name: Prepare the release
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: bundle exec rake release:prepare
+      - name: Create pull Request
```
Comment (ekohl) [inline .github/workflows/prepare_release.yml:73]: Looking at https://github.com/voxpupuli/puppet-example/pull/82 I wonder if we can come up with some better instructions in the body. For example, it could link to https://voxpupuli.org/docs/releasing_version/ though that isn't really explaining it well for this new process.

Comment (bastelfreak) [reply .github/workflows/prepare_release.yml:73]: My idea was to link to  https://voxpupuli.org/docs/releasing_version/ and update it

**Comment by bastelfreak** [comment]
this is now working fine and ready to be merged: https://github.com/voxpupuli/puppet-example/pull/97

**Review by sebastianrakel — APPROVED** [review]
